### PR TITLE
Support JSON API Spec 1.0 (we were hooked on RC3)

### PR DESCRIPTION
**Pull request by Boubalou**

### CODE REVIEW

File: dist/ember-json-api.js [hunk 1]
```diff
@@ -307,9 +307,9 @@ define("json-api-adapter",
        * Extract top-level "data" containing a single primary data
        */
       extractSingleData: function(data, payload) {
-        if(data.links) {
-          this.extractRelationships(data.links, data);
-          //delete data.links;
+        if(data.relationships) {
```
Comment (loboulet) [inline dist/ember-json-api.js:310]: Missing white space between the `if` and the `(`

File: dist/ember-json-api.js [hunk 2]
```diff
@@ -307,9 +307,9 @@ define("json-api-adapter",
...
+          this.extractRelationships(data.relationships, data);
+          //delete data.relationships;
```
Comment (loboulet) [inline dist/ember-json-api.js:312]: Shouldn't we remove this comment?

File: dist/ember-json-api.js [hunk 3]
```diff
@@ -322,9 +322,9 @@ define("json-api-adapter",
         var type = data.length > 0 ? data[0].type : null;
         var serializer = this;
         data.forEach(function(item) {
-          if(item.links) {
-            serializer.extractRelationships(item.links, item);
-            //delete data.links;
+          if(item.relationships) {
```
Comment (loboulet) [inline dist/ember-json-api.js:325]: Missing white space between the `if` and `(`

File: dist/ember-json-api.js [hunk 4]
```diff
@@ -339,33 +339,33 @@ define("json-api-adapter",
         var models = {};
         var serializer = this;
 
-        sideloaded.forEach(function(link) {
-          var type = link.type;
-          if(link.links) {
-            serializer.extractRelationships(link.links, link);
+        sideloaded.forEach(function(relationship) {
+          var type = relationship.type;
+          if(relationship.relationships) {
```
Comment (loboulet) [inline dist/ember-json-api.js:344]: Same issue with `if` and `(`

File: src/json-api-serializer.js [hunk 5]
```diff
@@ -138,33 +138,33 @@ DS.JsonApiSerializer = DS.RESTSerializer.extend({
...
-    resource.links = {};
+    resource.relationships = {};
 
-    for (link in links) {
-      association = links[link];
-      link = Ember.String.camelize(link.split('.').pop());
+    for (relationship in relationships) {
+      association = relationships[relationship];
+      relationship = Ember.String.camelize(relationship.split('.').pop());
       if(!association) { continue; }
```
Comment (loboulet) [inline src/json-api-serializer.js:168]: Missing white space between the `if` and `(`

File: src/json-api-serializer.js [hunk 6]
```diff
@@ -138,33 +138,33 @@ DS.JsonApiSerializer = DS.RESTSerializer.extend({
     var models = {};
     var serializer = this;
 
-    sideloaded.forEach(function(link) {
-      var type = link.type;
-      if(link.links) {
-        serializer.extractRelationships(link.links, link);
+    sideloaded.forEach(function(relationship) {
+      var type = relationship.type;
+      if(relationship.relationships) {
```
Comment (loboulet) [inline src/json-api-serializer.js:143]: Missing white space between the `if` and `(`

File: src/json-api-serializer.js [hunk 7]
```diff
@@ -121,9 +121,9 @@ DS.JsonApiSerializer = DS.RESTSerializer.extend({
     var type = data.length > 0 ? data[0].type : null;
     var serializer = this;
     data.forEach(function(item) {
-      if(item.links) {
-        serializer.extractRelationships(item.links, item);
-        //delete data.links;
+      if(item.relationships) {
```
Comment (loboulet) [inline src/json-api-serializer.js:124]: Missing white space between the `if` and `(`

File: src/json-api-serializer.js [hunk 8]
```diff
@@ -106,9 +106,9 @@ DS.JsonApiSerializer = DS.RESTSerializer.extend({
    * Extract top-level "data" containing a single primary data
    */
   extractSingleData: function(data, payload) {
-    if(data.links) {
-      this.extractRelationships(data.links, data);
-      //delete data.links;
+    if(data.relationships) {
```
Comment (loboulet) [inline src/json-api-serializer.js:109]: Missing white space between the `if` and `(`

**Comment by Boubalou** [comment]
Don't know if you saw it, but all I did was to rename most of the variables, I didn't touch the previous code, including commented out stuff. There is far more work to do for it to comply on eslint, which its not right now and I don't think we will ever do it, its bound to deprecate fast enough.

**Comment by loboulet** [comment]
:+1:

**Pull request merged**